engine/src/eth/retry_rpc.rs

```diff
@@ -306,9 +309,43 @@ impl<Rpc: EthSigningRpcApi> EthersRetrySigningRpcApi for EthRetryRpcClient<Rpc>
 								estimated_gas.saturating_mul(U256::from(4u64)) / 3u64
 							},
 						});
+						Ok(transaction_request)
+					})
+				}),
+				log.clone(),
+				MAX_BROADCAST_RETRIES,
+			)
+			.await?;
+
+		// Sign the transaction
+		let tx = self
+			.rpc_retry_client
+			.request_with_limit(
+				Box::pin(move |client| {
+					let transaction_request = transaction_request.clone();
+					#[allow(clippy::redundant_async_block)]
+					Box::pin(async move {
+						client
+							.sign_transaction(transaction_request)
+							.await
+							.context("Failed to Sign ETH transaction")
+					})
+				}),
+				log.clone(),
+				// Only sign the transaction once
+				0,
```

Comment:
should be 1 right? - technically don't think it matters as it'll short circuit at the same place - but 1 would be more inline with the `MAX_BROADCAST_RETRIES`

Reply:
also maybe a const rather than magic number

Reply:
I assume that what you mean that it will only retry if it has failed to sign. If we are sure that there is no way it can retry signing unless it has failed before, then I'm also fine with using `MAX_BROADCAST_RETRIES`. However, doing it only once is the safest way to make sure it's not double signed and this is unrelated to the broadcast retries, so I think leaving it as is is fine.

Reply:
In this case, it's unrelated to the number of tries, it'd still be 1 try here is the point of my comment - more just how we represent that in this particular case.